plugins/wm-actions/wm-actions.cpp

```diff
@@ -0,0 +1,192 @@
+            }
+        }
+    };
+
+    /**
+     * Ensures views marked as above are still above
+     * if their output changes.
+     */
+    wf::signal_connection_t on_view_output_changed
+    {[=] (wf::signal_data_t *data)
+        {
+            auto signal = static_cast<wf::view_moved_to_output_signal*>(data);
+            if (signal->new_output != output)
+            {
+                return;
+            }
+
+            auto view = signal->view;
+
+            if (!view)
+            {
+                return;
+            }
+
+            if (view->has_data("wm-actions-above"))
+            {
+                output->workspace->add_view_to_sublayer(view, always_above);
+            }
+        }
+    };
+
+    /**
+     * Ensures views marked as above are still above
+     * if they are minimized and unminimized.
+     */
+    wf::signal_connection_t on_view_minimized
+    {[=] (wf::signal_data_t *data)
+        {
+            auto signal = static_cast<wf::view_minimize_request_signal*>(data);
+            auto view   = signal->view;
+
+            if (!view)
+            {
+                return;
+            }
+
+            if (view->get_output() != output)
+            {
+                return;
+            }
+
+            if (view->has_data("wm-actions-above") && (signal->state == false))
+            {
+                output->workspace->add_view_to_sublayer(view, always_above);
+            }
+        }
+    };
+
+    /**
+     * The default activator bindings.
+     */
+    wf::activator_callback on_toggle_above =
+        [=] (wf::activator_source_t source, uint32_t) -> bool
+    {
+        auto view = choose_view(source);
+
+        return toggle_keep_above(view);
+    };
+
+    wf::activator_callback on_toggle_fullscreen =
+        [=] (wf::activator_source_t source, uint32_t) -> bool
+    {
+        if (!output->can_activate_plugin(this->grab_interface))
+        {
+            return false;
+        }
+
+        auto view = choose_view(source);
+        if (!view || (view->role != wf::VIEW_ROLE_TOPLEVEL))
+        {
+            return false;
+        }
+
+        view->fullscreen_request(view->get_output(), !view->fullscreen);
+
+        return true;
+    };
+
+  public:
+    void init() override
+    {
+        always_above = output->workspace->create_sublayer(
+            wf::LAYER_WORKSPACE, wf::SUBLAYER_DOCKED_ABOVE);
+        output->add_activator(toggle_above, &on_toggle_above);
+        output->add_activator(toggle_fullscreen, &on_toggle_fullscreen);
+        output->connect_signal("wm-actions-view-above", &on_toggle_above_signal);
+        output->connect_signal("view-minimized", &on_view_minimized);
+        wf::get_core().connect_signal("view-moved-to-output",
+            &on_view_output_changed);
+    }
+
+    void fini() override
+    {
+        auto always_on_top_views =
+            output->workspace->get_views_in_sublayer(always_above);
+
+        for (auto it = std::begin(always_on_top_views);
```

Comment:
`for (auto view : always_on_top_views)` is shorter, and you don't have to deal with iterators at all.